packages/@angular/cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,42 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+const execPromise = Promise.denodeify(exec);
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return Promise
+      .all([checkYarn(), checkCNPM()])
+      .then((data: Array<boolean>) => {
+        if (packageManager === 'npm') {
```

Comment:
How is user asking? If I have yarn/cnpm installed I just check for any and use whichever I found. Use just does `ng new foo`